Source/WebCore/platform/mac/ThemeMac.mm

```diff
@@ -408,74 +408,6 @@ static bool drawCellFocusRing(NSCell *cell, NSRect cellFrame, NSView *controlVie
     return false;
 }
 
-static void paintToggleButton(ControlPartType buttonType, ControlStates& controlStates, GraphicsContext& context, const FloatRect& zoomedRect, float zoomFactor, ScrollView* scrollView, float deviceScaleFactor)
-{
-    BEGIN_BLOCK_OBJC_EXCEPTIONS
-
-    RetainPtr<NSButtonCell> toggleButtonCell = static_cast<NSButtonCell *>(controlStates.platformControl());
-    IntSize zoomedRectSize = IntSize(zoomedRect.size());
-
-    if (controlStates.isDirty()) {
-        if (!toggleButtonCell)
-            toggleButtonCell = createToggleButtonCell(buttonType);
-        configureToggleButton(toggleButtonCell.get(), buttonType, controlStates, zoomedRectSize, zoomFactor, true);
-    } else {
-        if (!toggleButtonCell) {
-            if (buttonType == ControlPartType::Checkbox)
-                toggleButtonCell = sharedCheckboxCell(controlStates, zoomedRectSize, zoomFactor);
```

Comment:
We should think about how to implement `inflateControlPaintRect` in terms of the new control architecture, so that we can share more code, and remove `sharedCheckboxCell`, `sharedRadioCell`, and related methods from `ThemeMac`.

Could also be done in a separate patch, once all buttons are ready.